File: frontend/src/services/dashboardService.js
import { apiClient } from './api';
import { SIMILAR_EVENTS } from '../data/dashboardData';

export const getDashboardData = async () => {
  try {
    const data = await apiClient.get('/dashboard');
    return {
      similarEvents: SIMILAR_EVENTS, // keep mock similar events
      current_solar_status: data.current_solar_status || 'Unknown',
      mission_status: data.mission_status || 'Unknown',
      hardening_index: data.hardening_index || 0,
      last_update_time: data.last_update_time || '',
      forecast_probability: data.forecast_probability || 0,
      active_alert: !!data.active_alert
    };
  } catch (error) {
    throw error;
  }
};

export default {
  getDashboardData
};

File: backend/services/dashboard_service.py
import pandas as pd
from config import DATA_DIR

def get_dashboard_data():
    features_path = DATA_DIR / "processed" / "features.csv"
    forecast_path = DATA_DIR / "processed" / "forecast_outputs.json"
    
    result = {
        "current_solar_status": "Active",
        "mission_status": "GOES-16 Online",
    }
    
    # Consume pipeline features output
    if features_path.exists():
        try:
            df = pd.read_csv(features_path)
            last_row = df.iloc[-1]
            result["hardening_index"] = float(last_row.get("spectral_hardening_ratio", 1.0))
            result["last_update_time"] = str(last_row.get("time", "Unknown"))
        except Exception:
            pass
            
    if "hardening_index" not in result:
        result.update({
            "status": "error",
            "message": "Pipeline output missing",
            "missing_file": str(features_path),
            "hardening_index": 1.14,
            "last_update_time": "2026-06-03T12:00:00Z"
        })
        
    if forecast_path.exists():
        import json
        with open(forecast_path, "r") as f:
            forecast = json.load(f)
            result["forecast_probability"] = forecast.get("flare_probability", 0.0)
            result["active_alert"] = forecast.get("flare_probability", 0.0) > 0.6
    else:
        result["forecast_probability"] = 0.45
        result["active_alert"] = False

    return result
